Feature/implement character details screen

Adam0091 opened the pull request:
1. Добавлена страница "CharacterPage"
2. Изменены пути alies
3. Deploy: https://adam0091.github.io/rick-and-morty/
4. Done: 04.08.2022

Adam0091 committed:
fix: change paths

Adam0091 committed:
feat: add CharacterPage
 and InformationItem component

Adam0091 committed:
fix: change config for craco

Adam0091 committed:
fix: change craco config
and add homepage in package.json

Adam0091 committed:
fix: add  height: 100% for CharacterWrapper

Adam0091 committed:
feat: add type for CharacterPage

Adam0091 committed:
fix: rules eslint

Impeqq commented on src/components/CharactersPage/CharactersList/CharactersList.tsx:
```diff
@@ -19,7 +19,9 @@ export const CharactersList = ({ characters, visible }: TProps) => {
     <CharactersWrapper>
       {characters.slice(0, visible).map((character: CharacterType) => (
-        <CharacterCard key={character.id} character={character} />
+        <StyledLink to={`/character/${character.id}`} key={character.id}>
```
нужно брать роут из какой-то переменной, чтобы было например `${routePath.characterPage}/${character.id}`

Impeqq commented on src/components/Header/Header.tsx:
```diff
@@ -27,7 +28,7 @@ export const Header = () => {
             <ListStyle isActive={isActive} onClick={handlerBurger}>
               <li>
-                <NavLink to="">Characters</NavLink>
+                <NavLink to="characters">Characters</NavLink>
```
нужно брать роут из какой-то переменной, чтобы было например routePath.characters

Impeqq commented on src/containers/СharacterPage/CharacterPage.tsx:
```diff
@@ -0,0 +1,168 @@
+            <ul className="informations__list">
+              {Object.keys(infoCharacter).map((key) =>
+                infoCharacter[key].isLink ? (
+                  <StyledLink to={infoCharacter[key].path as string} key={key}>
```
to= принимает string | undefined, у тебя path и так string | undefined, as string лишний

Adam0091 replied:
Тогда я немного в замешательстве 
![image](https://user-images.githubusercontent.com/55671230/182906849-7d3f172e-7459-477b-91c1-fc9a0fb7ddaa.png)

Impeqq commented on src/routes/routesConfig.tsx:
```diff
@@ -0,0 +1,19 @@
+    element: <CharactersPage />,
+  },
+  {
+    path: "/character/:id",
```
а тут например использовать path: `${routePath.character}/:id`

Impeqq commented on src/containers/СharacterPage/CharacterPage.style.ts:
```diff
@@ -0,0 +1,157 @@
+  width: 24px;
+  height: 24px;
+
+  background-image: url("${arrowNext}");
```
давай не использовать background-image вообще. И еще это же кнопка, стоит сделать `<button><img/></button>`

Impeqq commented on src/components/Header/Header.tsx:
```diff
@@ -27,7 +28,7 @@ export const Header = () => {
           <NavContainer>
```
это должен быть не div а nav

Impeqq commented on src/containers/СharacterPage/CharacterPage.style.ts:
```diff
@@ -0,0 +1,157 @@
+  }
+`;
+
+export const CharacterTitle = styled.div`
```
если внутри элемента текст то используем span

Adam0091 committed:
fix: wrappe path into constants,
replace background-image with img tag,
replace divs with span,
in the style of the NavContainer component
tag is replace from div to nav
